okio/src/main/java/okio/BufferedSource.java

```diff
@@ -173,6 +173,20 @@
   String readUtf8LineStrict() throws IOException;
 
   /**
+   * Removes and returns a single UTF-8 code point, reading between 1 and 4 bytes as necessary.
+   *
+   * <p>If this source is exhausted before a complete code point can be read, this throws an {@link
+   * java.io.EOFException} and consumes no input.
```

Comment:
Nice doc. I don't think we're very good about documenting the behavior when methods throw (related to the contents of the underlying buffer).